mesonbuild/modules/windows.py

```diff
@@ -62,7 +68,22 @@ def _find_resource_compiler(self, state):
         if not rescomp.found():
             raise MesonException('Could not find Windows resource compiler')
 
-        self._rescomp = rescomp
+        rescomp_type = None
+        for (arg, match, type) in [
+                ('/?', '^.*Microsoft.*Resource Compiler.*$', ResourceCompilerType.rc),
+                ('--version', '^GNU windres.*$', ResourceCompilerType.windres),
+        ]:
+            p, o, e = mesonlib.Popen_safe(rescomp.get_command() + [arg])
+            m = re.search(match, o, re.MULTILINE)
+            if m:
+                mlog.log('Windows resource compiler: %s' % m.group())
+                rescomp_type = type
+                break
```

Comment:
Why not just set `self._rescomp` here and return, then the raise can be unconditional when leaving the loop?

Reply:
I guess I find it clearer to have exceptions as diverging from the normal flow of execution, which terminates at the end of the function.

But you're right that I can write 'for:/else:' to avoid a comparison whose result is statically known, and avoid an unnecessary variable.

Reply:
I like `for/else` statements myself, but usually other people don't so I don't recommend them :)

Reply:
It doesn't really matter though, don't hold up merging on account of this.